### PR TITLE
BIP39 derived seed for a BIP84 wallet

### LINKED ISSUE #68: Create BIP-39 compatible BDK wallet
Upon the first start of the application, a new bitcoin wallet should be created from with a seed compatible with BIP39.
This should allow us to export the seed words.

### PR DESCRIPTION
**Pull request by holzeis**
This PR derives a BIP84 wallet from a BIP39 seed phrase and exposes a corresponding rust api through the flutter rust bridge. Additionally this PR already improves the structuring (to the best of my limited flutter skills).

resolves #68 #11 

This PR is still draft as the seed screen needs a little bit more care.

### CODE REVIEW

**Comment by holzeis** [comment]
Actually, that PR is already quite big and there are a lot of useful things already in there. I've created a follow up ticket to build the seed widget. https://github.com/itchysats/10101/issues/84.

File: rust/Cargo.toml [hunk 1]
```diff
@@ -10,3 +10,8 @@ crate-type = ["cdylib", "staticlib"] # Android needs cdylib, whereas iOS staticl
 [dependencies]
 anyhow = { version = "1", features = ["backtrace"] }
 flutter_rust_bridge = "1.49.0"
+bip39 = "1.0.1"
```
Comment (klochowicz) [inline rust/Cargo.toml:13]: I suspect you can just the version integrated with `bdk` instead by enabling the relevant feature flag? as a bonus, it enables to nicely create descriptor wallets from it (so your code would get simpler too!)

File: rust/Cargo.toml [hunk 2]
```diff
@@ -10,3 +10,8 @@ crate-type = ["cdylib", "staticlib"] # Android needs cdylib, whereas iOS staticl
...
+rand = "^0.6.0"
+hkdf = "0.12"
+sha2 = "0.10"
+bdk = "0.23.0"
```
Comment (klochowicz) [inline rust/Cargo.toml:17]: note to self: I forgot to enable dprint for toml files 🤦‍♂️ should be fixed in a sec :)

**Review by klochowicz — COMMENTED** [review]
nice start! I haven't look much into flutter here, I was focusing on Rust feedback

**Comment by da-kami** [comment]
Given that @klochowicz is working on figuring out what is best to achieve libaray API calls into ruest, I would propose that we don't implement complete features where we touch the UI *and* the rust backend, but we implement either UI or backend.
I think the initial focus should be on the UI, because the UI as consumer should drive the API design for what is needed in rust. I propose that we focus on the when working on issues like #85 and #84, and plug in the rust code later. We *can* discuss the API while working on such tickets and then ramp up the rust code in parallel.

**Comment by da-kami** [comment]
> [snip]
> This PR is still draft as the seed screen needs a little bit more care.

It's not marked as Draft PR on GiHub, is that on purpose?

File: rust/src/wallet.rs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+pub const MAINNET_ELECTRUM: &str = "ssl://blockstream.info:700";
+pub const TESTNET_ELECTRUM: &str = "ssl://blockstream.info:993";
+
+#[derive(Debug)]
+pub struct WalletInfo {
+    pub phrase: Vec<String>,
+}
+
+pub fn init_wallet(network: &str) -> Result<WalletInfo> {
```
Comment (da-kami) [inline rust/src/wallet.rs:21]: For me it is unclear if this API is what we actually want. It feels awkward to instantiate an Electrum client on every call, I think we need different lifecycle management here. 

@klochowicz is working on the lifecycle and state management of the rust components within Dart. I would focus on the UI and not try to achieve the *actual* backend call yet but plug in dummy values for now. 

You could even go as far as implementing a value provider (e.g. factory) that can be switched from dummy value to an actual API call easily.

**Review by da-kami — APPROVED** [review]
Happy to merge this PR, but I am a bit confused that this is the first thing we worked on. I would have expected a PR that adds the Dashboard with dummy values similar to what is mocked in the wireframe and described in #85 and only then wire in API calls.

File: rust/Cargo.toml (hunk 2 above)
Comment (klochowicz) [reply rust/Cargo.toml:17]: this is fixed, you can rebase and run `dprint fmt` on it :)

File: rust/src/wallet.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::wallet;
+
+    #[test]
+    fn init_wallet_success() {
+        wallet::init_wallet("bitcoin").expect("wallet to be initialized");
```
[the remark this answers is not in the document]
Comment (holzeis) [reply rust/src/wallet.rs:56]: yes. it's defined by `bdk::bitcoin::Network`

**Review by da-kami — APPROVED** [review]

File: rust/src/wallet.rs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+use bdk::Wallet;
+use std::str::FromStr;
+
+pub const MAINNET_ELECTRUM: &str = "ssl://blockstream.info:700";
+pub const TESTNET_ELECTRUM: &str = "ssl://blockstream.info:993";
+
+#[derive(Debug)]
+pub struct WalletInfo {
+    pub phrase: Vec<String>,
+}
```
[the remark this answers is not in the document]
Comment (holzeis) [reply rust/src/wallet.rs]: removed that code.

**Comment by holzeis** [comment]
Thanks @klochowicz @da-kami  for the comments. The rest of them will be addressed in follow up PRs.

**Pull request merged**